Dockerfile

```diff
@@ -1,34 +1,36 @@
 # Simple usage with a mounted data directory:
 # > docker build -t gaia .
-# > docker run -v $HOME/.gaiad:/root/.gaiad gaia init
-# > docker run -v $HOME/.gaiad:/root/.gaiad gaia start
-
-FROM alpine:edge
+# > docker run -it -p 46657:46657 -p 46656:46656 -v ~/.gaiad:/root/.gaiad -v ~/.gaiacli:/root/.gaiacli gaia gaiad init
+# > docker run -it -p 46657:46657 -p 46656:46656 -v ~/.gaiad:/root/.gaiad -v ~/.gaiacli:/root/.gaiacli gaia gaiad start
+FROM golang:alpine AS build-env
 
 # Set up dependencies
-ENV PACKAGES go glide make git libc-dev bash
-
-# Set up GOPATH & PATH
-
-ENV GOPATH       /root/go
-ENV BASE_PATH    $GOPATH/src/github.com/cosmos
-ENV REPO_PATH    $BASE_PATH/cosmos-sdk
-ENV WORKDIR      /cosmos/
-ENV PATH         $GOPATH/bin:$PATH
-
-# Link expected Go repo path
+ENV PACKAGES make git libc-dev bash
 
-RUN mkdir -p $WORKDIR $GOPATH/pkg $ $GOPATH/bin $BASE_PATH
+# Set working directory for the build
+WORKDIR /go/src/github.com/cosmos/cosmos-sdk
 
 # Add source files
-
-ADD . $REPO_PATH
+COPY . .
```

Comment:
I still don't get why the source needs to be copied into a container to build it. You could literally run the golang container with `-v` and `make build` to build binaries. Ask @cwgoes , he has a similar line for deterministic builds.

Reply:
Here:
`docker run --rm -it -v $(pwd):/go/src/github.com/tendermint/tendermint golang:1.10.2 make -C /go/src/github.com/tendermint/tendermint get_tools get_vendor_deps build` Change it for gaiad.

Reply:
We are copying the source code in to build in a consistent environment. This is the way things are normally done with docker builds imo. The two are equivalent, but this one encodes all that command ugliness in the `Dockerfile` in one simple line and lets us avoid pushing logic to the `Makefile` so I would say lets go with this.